src/worker.js

```diff
@@ -10,41 +10,26 @@ import { isConfigAtHomeRoot } from './is-config-at-home-root'
 
 process.title = 'linter-eslint helper'
 
-const ignoredMessages = [
-  // V1
-  'File ignored because of your .eslintignore file. Use --no-ignore to override.',
-  // V2
-  'File ignored because of a matching ignore pattern. Use --no-ignore to override.',
-  // V2.11.1
-  'File ignored because of a matching ignore pattern. Use "--no-ignore" to override.',
-  // supress warning that the current file is ignored by eslint by default
-  'File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern \'!<relative'
-    + '/path/to/filename>\'") to override.',
-  'File ignored by default. Use "--ignore-pattern \'!node_modules/*\'" to override.',
-  'File ignored by default. Use "--ignore-pattern \'!bower_components/*\'" to override.',
-]
+function lintJob({ cliEngineOptions, contents, eslint, filePath }) {
+  const cliEngine = new eslint.CLIEngine(cliEngineOptions)
 
-function lintJob(argv, contents, eslint, configPath, config) {
-  const noProjectConfig = (configPath === null || isConfigAtHomeRoot(configPath))
-  if (noProjectConfig && config.disableWhenNoEslintConfig) {
-    return []
-  }
-  eslint.execute(argv, contents)
-  return global.__LINTER_ESLINT_RESPONSE
-    .filter(e => !ignoredMessages.includes(e.message))
+  return typeof contents === 'string'
+    ? cliEngine.executeOnText(contents, filePath)
+    : cliEngine.executeOnFiles([filePath])
 }
 
-function fixJob(argv, eslint) {
-  const exit = eslint.execute(argv)
-  if (exit === 0) {
+function fixJob({ cliEngineOptions, contents, eslint, filePath }) {
+  const report = lintJob({ cliEngineOptions, contents, eslint, filePath })
+
+  eslint.CLIEngine.outputFixes(report)
+
+  if (!report.results.length || !report.results[0].messages.length) {
```

Comment:
Perhaps we can move to `executeOnText` for fix jobs at some point and turn off `warnIgnored`, as you mentioned in https://github.com/AtomLinter/linter-eslint/pull/873#discussion_r110076641, but for now let's keep the `ignoredMessages` and use them here.